### DISCUSSION TITLE
Use of sliced scroll or search after

### DISCUSSION THREAD

**Discussion by jry25** [ruflin/Elastica#1969] [Ideas]
Hi there. Today I wanted to add a feature to export documents to csv.

As this action is asynchronous and look for a lot of documents (more than 10k) I wanted to use a [sliced scroll](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html#slice-scroll). Unfortunately the slice option is not supported.

Is this kind of feature interesting? Should I work on a PR?

FYI: I tried to use sliced scroll as iteration over scroll results was extremely painfully slow.
```php
//prepare scroll
//loop over scroll
foreach ($scroll as $resultSet) {
    foreach($resultSet->getResults() as $result) {
        yield $result->getSource(); // 5 minutes between each yield
    }
}
```

**Comment by jry25** [comment]
I might have to switch to a [search after](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html#search-after) don't you think it's better?

> We no longer recommend using the scroll API for deep pagination. If you need to preserve the index state while paging through more than 10,000 hits, use the search_after parameter with a point in time (PIT).

Didn't pay attention to this warning.

**Reply by jry25** [reply]
Looks like I'm not the only one https://github.com/ruflin/Elastica/issues/1645

**Comment by jry25** [comment]
Unfortunately for me my ES server is on 7.9 version.
And right now I have an issue when I loop over a scroll

```
$count = 1;
foreach ($scroll as $scrollId => $resultSet) {
    foreach ($resultSet->getResults() as $result) {
        $this->logger->info(
            'Loop {count} with "{docsCount}" documents over scroll {scrollId}.',
            [
                'count' => $count,
                'docsCount' => $resultSet->count(),
                'scrollId' => $scrollId,
            ]
        );
        yield $result->getData();
    }
    ++$count;
}
$scroll->clear();
```
But the iteration stops close to the ends. I have 9757 docs matching my search and it stops on loop 824.
The scroll expiry time is set to 10m but it doesn't take 10m to loop. 
Do you have any idea why this problem occurs?

**Reply by ruflin** [reply]
Hi, didn't check your code in detail yet. One way of debugging these issues is often writing the raw ES query and check how it behaves in ES directly to better understand where the issue happens. Any chance you could try this?
